near-sdk/src/environment/mock/receipt.rs

```diff
@@ -1,47 +1,161 @@
-use crate::{AccountId, Gas, NearToken, PublicKey};
+use near_primitives_core::types::GasWeight;
+use near_vm_runner::logic::mocks::mock_external::MockAction as LogicMockAction;
+use near_vm_runner::logic::types::ReceiptIndex;
+
+use crate::{AccountId, Gas, NearToken};
 
 #[derive(Clone, Debug, PartialEq, Eq)]
 #[non_exhaustive]
 pub struct Receipt {
     pub receiver_id: AccountId,
-    pub actions: Vec<VmAction>,
+    pub receipt_indices: Vec<ReceiptIndex>,
+    pub actions: Vec<MockAction>,
 }
 
-#[derive(Clone, Debug, PartialEq, Eq)]
-#[non_exhaustive]
-pub enum VmAction {
-    CreateAccount,
+#[derive(serde::Serialize)]
+#[serde(remote = "GasWeight")]
+struct GasWeightSer(u64);
+
+#[derive(Debug, Clone, serde::Serialize, PartialEq, Eq)]
+pub enum MockAction {
```

Comment:
I am trying to understand if `receipt_index` is actually necessary at this level, and if the types mismatch with the blockchain is justified... @dj8yfo What is your understanding here?

@austinabell Can you shed some light here by any chance?

Reply:
I can't remember the context, and I don't know about the nearcore updates. I can think about it more another time, but I wouldn't wait on my response, sorry!